### PR TITLE
Dependency convergence + jakarta libraries.

### PR DESCRIPTION
**Pull request by tomas-langer**
Signed-off-by: Tomas Langer <tomas.langer@oracle.com>

Switched to `jakarta` group id for Jakarta EE standards (keeping `javax` package).
Fixed all related issues with imports, exclusions etc.
Added dependency convergence plugin to Helidon project.
Added sanity checks for MP - both classpath and module path to `build.sh`

Also resolved problems with MP JPMS - quickstart should now work with JPMS without issues.

Marking as API change, as the libraries we depend on have changed.

### CODE REVIEW

**Comment by tomas-langer** [comment]
Merged latest master (conflict)

File: grpc/client/src/test/java/io/helidon/grpc/client/GrpcChannelsProviderIT.java [hunk 1]
```diff
@@ -37,13 +37,15 @@
...
 import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 import services.TreeMapService;
 
 import static org.hamcrest.CoreMatchers.equalTo;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 
+@Disabled("Certificate expired on 27th March 2020")
```
Comment (romain-grecourt) [inline grpc/client/src/test/java/io/helidon/grpc/client/GrpcChannelsProviderIT.java:48]: The certificate has been updated right? If so this change should not be needed.

File: pom.xml [hunk 2]
```diff
@@ -747,6 +766,22 @@
                 <artifactId>reactive-streams-tck-flow</artifactId>
                 <version>${version.lib.reactivestreams}</version>
             </dependency>
+            <dependency>
```
Comment (romain-grecourt) [inline pom.xml:769]: These dependency management for 4th party dependencies, they belong in helidon-dependencies

Comment (tomas-langer) [reply pom.xml:769]: These are only internal dependencies (`reactive-streams-tck-flow`, `wagon-provider-api`, `hamcrest-library`, `weld-se-core`)
I will move the others to `dependencies`

Comment (romain-grecourt) [reply pom.xml:763]: just to make sure we are on the same page: anything that manages the version of a dependency that may end-up under target/libs for an helidon app needs to be defined in helidon-dependencies.

File: microprofile/jwt-auth/src/main/java/io/helidon/microprofile/jwt/auth/JwtAuthProvider.java [hunk 3]
```diff
@@ -610,6 +610,13 @@ private InputStream locateStream(String uri) throws IOException {
             InputStream is;
 
             URL url = Thread.currentThread().getContextClassLoader().getResource(uri);
+            if (url == null) {
```
Comment (romain-grecourt) [inline microprofile/jwt-auth/src/main/java/io/helidon/microprofile/jwt/auth/JwtAuthProvider.java:613]: Why is this change needed for this PR ?
Should this go in a separate PR ?

Comment (tomas-langer) [reply microprofile/jwt-auth/src/main/java/io/helidon/microprofile/jwt/auth/JwtAuthProvider.java:613]: Because this failed as part of the `module-path` based tests, and I needed to fix it. The build does not pass without it.
